lib/Test2/Compare/Negatable.pm

```diff
@@ -0,0 +1,119 @@
+package Test2::Compare::Negatable;
+use strict;
+use warnings;
+
+require overload;
+require Test2::Util::HashBase;
+
+sub import {
+    my ($pkg, $file, $line) = caller;
+
+    my $sub = eval <<"    EOT" or die $@;
+package $pkg;
+#line $line "$file"
+sub { overload->import('!' => 'clone_negate', fallback => 1); Test2::Util::HashBase->import('negate')}
+    EOT
+
+    $sub->();
+
+    no strict 'refs';
+    *{"$pkg\::clone_negate"} = \&clone_negate;
+    *{"$pkg\::toggle_negate"} = \&toggle_negate;
+}
+
+sub clone_negate {
+    my $self = shift;
+    my $clone = $self->clone;
+    $clone->toggle_negate;
+    return $clone;
+}
+
+sub toggle_negate {
+    my $self = shift;
+    $self->set_negate(!$self->negate ? 1 : 0);
```

Comment:
This is a toggle, not a setter, which means flip between possible values, if it is on it becomes off, vice versa, so this ternary is correct.